src/interfaces/earnings.spec.ts

```diff
@@ -0,0 +1,213 @@
+        getBlockNumber: getBlockNumberMock
+      }
+    }
+  }))
+}));
+
+describe("EarningsInterface", () => {
+  let earningsInterface: EarningsInterface<1>;
+
+  let mockedYearn: Yearn<ChainId>;
+
+  beforeEach(() => {
+    mockedYearn = new (Yearn as jest.Mock<Yearn<ChainId>>)();
+    earningsInterface = new EarningsInterface(mockedYearn, 1, new Context({}));
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  describe("protocolEarnings", () => {
+    describe("when there is no data", () => {
+      beforeEach(() => {
+        subgraphFetchQueryMock.mockResolvedValue(undefined);
+      });
+
+      it("should return a string representing the value of BigNumber(0)", async () => {
+        const actualProtocolEarnings = await earningsInterface.protocolEarnings();
+
+        expect(actualProtocolEarnings).toEqual("0");
+      });
+    });
+
+    describe("when there is data", () => {
+      beforeEach(() => {
+        oracleGetPriceUsdcMock.mockResolvedValueOnce("2.5");
+        subgraphFetchQueryMock.mockResolvedValue({
+          data: {
+            vaults: [
+              {
+                latestUpdate: {
+                  returnsGenerated: new BigNumber(2).multipliedBy(10 ** 18)
+                },
+                token: {
+                  decimals: 18,
+                  id: "tokenId"
+                }
+              },
+              {
+                token: {
+                  decimals: 18,
+                  id: "tokenIdWithoutLatestUpdate"
+                }
+              }
+            ]
+          }
+        });
+      });
+
+      it("should return the protocol earnings", async () => {
+        const actualProtocolEarnings = await earningsInterface.protocolEarnings();
+
+        expect(actualProtocolEarnings).toEqual(new BigNumber(5).toFixed(0));
+      });
+    });
+  });
+
+  describe("assetEarnings", () => {
+    describe("when there is no data", () => {
+      beforeEach(() => {
+        subgraphFetchQueryMock.mockResolvedValue(undefined);
+      });
+
+      it("should throw", async () => {
+        try {
+          await earningsInterface.assetEarnings("0x001");
+        } catch (error) {
+          expect(error).toStrictEqual(new SdkError("No asset with address 0x001"));
+        }
+      });
+    });
+
+    describe("when there is data", () => {
+      beforeEach(() => {
+        getAddressMock.mockReturnValue("0x001");
+        oracleGetPriceUsdcMock.mockResolvedValueOnce("3.5");
+        subgraphFetchQueryMock.mockResolvedValue({
+          data: {
+            vault: {
+              token: {
+                id: "vaultTokenId",
+                decimals: 18
+              },
+              latestUpdate: {
+                returnsGenerated: new BigNumber(2).multipliedBy(10 ** 18)
+              }
+            }
+          }
+        });
+      });
+
+      it("should return the asset earnings", async () => {
+        const actualAssetEarnings = await earningsInterface.assetEarnings("0x001");
```

Comment:
could we use `getAddressMock` here instead of `0x001`

Reply:
This is the actual call to the `assetEarnings` function, so we should not rely on the return of a mock, because that function does not call the `getAddress()` -- if you want we can create a variable in the describe to store that address and check against it in the test. Extracted with https://github.com/yearn/yearn-sdk/pull/223/commits/e7be836e253ae69137a3160590fce91149d10169

If you see in the actual code, you have

`assetAddress: getAddress(assetAddress),`

So we would be mocking the call on the mock itself `getAddress(getAddressMock)`, so it does not seem reasonable to me